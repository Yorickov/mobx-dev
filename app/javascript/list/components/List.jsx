import React from 'react';
import Table from './Table';
import Controls from './Controls';
import tableStore from '../stores/TableStore';

const List = () => (
  <div>
    <h1>Sprint Board:</h1>
    <Controls store={tableStore} />
    <Table store={tableStore} />
  </div>
);

export default List;
