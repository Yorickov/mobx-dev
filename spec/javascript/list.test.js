import React from 'react';
import List from '../../app/javascript/list/components/List';

it('List', () => {
  const wrapper = mount(<List />);
  expect(wrapper.render()).toMatchSnapshot();

  const newTextInput = wrapper.find('.controls input');
  newTextInput.simulate('change', { target: { value: 'J' } });
  expect(wrapper.render()).toMatchSnapshot();

  const clearButton = wrapper.find('.controls button').first();
  clearButton.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});
