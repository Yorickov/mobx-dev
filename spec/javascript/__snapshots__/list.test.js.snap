// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List 1`] = `
<div>
  <h1>
    Sprint Board:
  </h1>
  <div
    class="controls"
  >
    <button>
      Clear table
    </button>
    <button>
      Add record
    </button>
    <input
      type="text"
      value=""
    />
  </div>
  <table>
    <thead>
      <tr>
        <td>
          Name:
        </td>
        <td>
          Points:
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jack
        </td>
        <td>
          12
        </td>
      </tr>
      <tr>
        <td>
          Max
        </td>
        <td>
          10
        </td>
      </tr>
      <tr>
        <td>
          Leo
        </td>
        <td>
          8
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td>
          Team Points:
        </td>
        <td>
          30
        </td>
      </tr>
      <tr>
        <td>
          Top Performer:
        </td>
        <td>
          Jack
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`List 2`] = `
<div>
  <h1>
    Sprint Board:
  </h1>
  <div
    class="controls"
  >
    <button>
      Clear table
    </button>
    <button>
      Add record
    </button>
    <input
      type="text"
      value="J"
    />
  </div>
  <table>
    <thead>
      <tr>
        <td>
          Name:
        </td>
        <td>
          Points:
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jack
        </td>
        <td>
          12
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td>
          Team Points:
        </td>
        <td>
          30
        </td>
      </tr>
      <tr>
        <td>
          Top Performer:
        </td>
        <td>
          Jack
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`List 3`] = `
<div>
  <h1>
    Sprint Board:
  </h1>
  <div
    class="controls"
  >
    <button>
      Clear table
    </button>
    <button>
      Add record
    </button>
    <input
      type="text"
      value="J"
    />
  </div>
  <table>
    <thead>
      <tr>
        <td>
          Name:
        </td>
        <td>
          Points:
        </td>
      </tr>
    </thead>
    <tbody />
    <tfoot>
      <tr>
        <td>
          Team Points:
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td>
          Top Performer:
        </td>
        <td />
      </tr>
    </tfoot>
  </table>
</div>
`;
